lib/src/null_safety_analysis.dart

```diff
@@ -113,15 +115,29 @@ class NullSafetyAnalysis {
       return NullSafetyAnalysisResult(NullSafetyCompliance.analysisFailed,
           'Could not resolve constraints: $e');
     }
+    return nullSafetyComplianceOfResolution(result);
+  }
 
+  /// Decides if all dependendencies (transitively) have a language version
+  /// opting in to null safety, and no files in lib/ of these packages opt out
+  /// to a pre-null-safety language version.
+  ///
+  /// This will download all dependencies into [cache].
+  ///
+  /// Assumes [mainPackage] is opted in.
+  Future<NullSafetyAnalysisResult> nullSafetyComplianceOfResolution(
+      SolveResult result) async {
```

Comment:
We could also be passing `root` as an argument if we wanted to, instead of making it public. I don't know what's right here.

Or rather, instead of passing `root` we could be passing a `package` argument, and rename the method to `nullSafetyComplianceInResolution`.. and then say it determines if a given _package_ is _null-safety compliant_ in a given _resolution_.

----
hmm, couldn't we just pass `Iterable<PackageId>` and ask if they are null-safety compliant?

Before we used:
```dart
final boundSource = dependencyId.source.bind(_systemCache);
final pubspec = await boundSource.describe(dependencyId);
```
So we don't need `result.pubspecs`, and we don't need to pass `root` or filter out the `dummy root package`, if we filter before we pass an `Iterable<PackageId>` to `nullSafetyComplianceOfResolution`..

Reply:
Done